File: datascope-ui/src/types/query-history.ts
export interface QueryHistory {
  id: string;
  sql: string;
  dataSourceId: string;
  dataSourceName: string;
  executionTime: number;
  affectedRows: number;
  status: 'SUCCESS' | 'FAILED';
  errorMessage?: string;
  parameters?: string;
  executionIp: string;
  createdBy: string;
  createdAt: string;
}

export interface QueryHistoryQuery {
  page: number;
  size: number;
  dataSourceId?: string;
  status?: string;
  startTime?: string;
  endTime?: string;
}

export interface QueryHistoryResponse {
  content: QueryHistory[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}

export interface QueryStats {
  totalQueries: number;
  successQueries: number;
  failedQueries: number;
  averageExecutionTime: number;
  maxExecutionTime: number;
  minExecutionTime: number;
  totalAffectedRows: number;
}

File: datascope-domain/src/main/java/com/datascope/domain/model/query/QueryHistory.java
package com.datascope.domain.model.query;

import com.datascope.common.model.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 查询历史实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class QueryHistory extends BaseEntity {
    /**
     * 查询ID
     */
    private String queryId;
    
    /**
     * 查询参数（JSON格式）
     */
    private String parameters;
    
    /**
     * 执行时间（毫秒）
     */
    private Long executionTime;
    
    /**
     * 影响行数
     */
    private Long affectedRows;
    
    /**
     * 状态（SUCCESS, FAILED）
     */
    private String status;
    
    /**
     * 错误信息
     */
    private String errorMessage;
    
    /**
     * 执行IP
     */
    private String executionIp;
}